### DISCUSSION TITLE
overleaf修改字体问题

### DISCUSSION THREAD

**Discussion by suppent** [note286/xduts#89] [Q&A] [answered]
\documentclass{xdupgthesis}
\xdusetup{
    style/cjk-font = none,
    style/math-font = none,
    style/latin-font = none,
    style/font-type = file,
}

请问大佬，中文字体已经从旧的模板中复制出来上传到overleaf了，在fonts文件夹下，但是编译后，不显示中文，是怎么回事？谢谢

**Comment by note286** [comment, marked as answer]
你设置为`none`当然显示不出来。

**Reply by suppent** [reply]
\documentclass{xdupgthesis}
\xdusetup{
    style/font-type = file
}

\begin{document}
你好
\end{document}
我刚才试了一下，只写这一行，还是编译不出来

**Reply by suppent** [reply]
> \documentclass{xdupgthesis}
> \xdusetup{
>     style/font-type = file
> }
> 
> \begin{document}
> 你好
> \end{document}
> 我刚才试了一下，只写这一行，还是编译不出来

只写这一行，编译出来的错误就是找不到字体

**Reply by note286** [reply]
你上传的字体文件要和字体配置相对应；来个star。

**Reply by suppent** [reply]
字体我是从那个旧版本的模板里直接复制过来的，我在那个pdf文档里也对应了一下，发现没有问题啊麻烦大佬帮忙看一下，我在这个地方看了好久了，谢谢！![Screenshot_2023-03-23-01-04-15-966_com.android.chrome.png](https://user-images.githubusercontent.com/21963812/226982386-43b83ff8-538d-44e6-a745-72aad86db40c.png)

**Reply by note286** [reply]
[note286/xduts: Xidian University TeX Suite 西安电子科技大学LaTeX套装](https://github.com/note286/xduts)和[note286/xdupgthesis: [停止维护 请使用note286/xduts]西安电子科技大学研究生学位论文XeLaTeX模板](https://github.com/note286/xdupgthesis)是完全独立的两个项目，二者没有任何关系。

你在指定好字体配置之后，需要将所有的字体文件均上传，字体文件可以见 https://github.com/note286/xduts/discussions/40#discussioncomment-5117827 。

![image](https://user-images.githubusercontent.com/16207292/227073505-47ca9b6c-f787-4acc-be82-ada4d85282bd.png)

![image](https://user-images.githubusercontent.com/16207292/227073594-1eb12d21-e994-42f2-83b1-712f6674ae78.png)

**Reply by note286** [reply]
`style/font-type=file`目前内部逻辑确实有点缺陷，需要再优化下，目前没有时间处理了，只能等几个月后再说了。

**Reply by suppent** [reply]
大佬，这个问题我还是没有解决（好菜。。。），我把字体上传以后，那个字体的配置只需要写
    style/font-type = file,
    style/font-path= {fonts}
这两行吗？我现在这样写的，编译出的字体依然没有变。

**Reply by note286** [reply]
我很确信我的手册不会让人误解，我觉得是你陷入了思维的误区出不来了。正确的逻辑是，选择合适的字体配置，然后选择字体文件名方式，并将所对应的字体文件全部上传。

```latex
\documentclass{xdupgthesis}
\xdusetup{
style/cjk-font   = win,
style/latin-font = tac,
style/math-font  = cambria,
style/font-type  = file,
}
\begin{document}
\end{document}
```

**Reply by suppent** [reply]
原来是这样，我一直理解的是上传字体后，指定字体类型及路径就好了，谢谢！！

**Reply by note286** [reply]
没那么智能，按照你的逻辑，我还需要去写检查你上传了什么字体文件的代码，累死我得了。

**Comment by suppent** [comment]
好的，我再试一下，谢谢大佬指点！
